tiledb/sm/cpp_api/group.h

```diff
@@ -473,13 +473,11 @@ class Group {
   std::string dump(const bool recursive) const {
     auto& ctx = ctx_.get();
     tiledb_ctx_t* c_ctx = ctx.ptr().get();
-    char* str;
+    tiledb_string_t* str;
     ctx.handle_error(
-        tiledb_group_dump_str(c_ctx, group_.get(), &str, recursive));
+        tiledb_group_dump_str_v2(c_ctx, group_.get(), &str, recursive));
```

Comment:
I really don't see that much into this code, but don't we need to free it here since we copy it into string? or does the convert_to_string logic frees the string?